### PR TITLE
Add threadlocker to infra

### PR DESCRIPTION
**Pull request by aboutdavid**
Why this exists: Threadlocker locks threads. Threads can become dumpsterfires. We lock the threads to prevent the spreading of dumpsterfires. I.e. a controversial confession causes a digital hateful flamewar, so the fire department locks it to make it die down.

How does thread locking works:
1. Click on the top-most message, go to message actions and click "Lock thread"
2. Enter in the details to the modal. The reason why and when it will unlock.
image
3. The details and thread ts get saved into the database
4. A log will be posted in #locked-threads (in future revisions the admin field was removed_
image
5. Threadlocker will watch for messages under that thread and delete them
6. The user will be informed via an ephemeral that the thread is locked
7. A setInterval will check every minute to see if the thread is unlocked OR another admin can unlock OR a message will trigger the unlocking mechanism if the time expired.

### CODE REVIEW

**Comment by jaspermayone** [comment]
@aboutdavid is this a request for a heroku project? github repo?

**Comment by aboutdavid** [comment]
no it's supposed to go in the repo so each pr can be reviewed by infra afaik

**Comment by jaspermayone** [comment]
totally missed this being a pr

**Comment by jaspermayone** [comment]
should this not be its own repo tho?

**Comment by jaspermayone** [comment]
i also know nothing dont listen to me lol

**Comment by grymmy** [comment]
Please add high level commentary to this PR describing what this tool is and exactly how it is expected to be used in practice.  I would also be good to provide a paragraph on how it functions (it appears to do some polling/etc) internally

**Comment by aboutdavid** [comment]
I updated the top level message with info, should I add more?

File: threadlocker/index.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    profilesSampleRate: 1.0,
+});
+
+
+(async () => {
+    async function autoUnlock() {
+        let span;
+        if (process.env.SENTRY_DSN) span = Sentry.startInactiveSpan({ name: "unlock_thread_cron" });
+        // Automatic unlocking, checks database once per minute
+        const threads = await prisma.thread.findMany({
```
Comment (grymmy) [inline threadlocker/index.js:30]: just wondering - will this list of locked threads grow and grow and grow potentially?  What is the lifetime of a locked thread?

Comment (grymmy) [reply threadlocker/index.js:30]: it sometimes will be like 9/9/9999 right?

**Review by grymmy — APPROVED** [review]
Seems pretty reasonable!

File: threadlocker/index.js (hunk 1 above)
Comment (grymmy) [reply threadlocker/index.js:30]: Nevermind.  Reading more closely, this should be a performant query regardless of # of locked threads.

**Pull request merged**